File: codility/MaxDoubleSliceSum.js
function solution(A) {
    // Implement your solution here
    const n = A.length;
    const larr = [];
    larr[0] = 0;
    larr[1] = 0;
    for (let i = 2; i < n-1; i++) {
        const a = A[i-1];
        larr[i] = Math.max(larr[i-1]+a, a, 0)
    }
    // console.log(larr)
    const rarr = [];
    rarr[n-1] = 0;
    rarr[n-2] = 0;
    for(let i = n-3 ; i > 0 ; i--){
        const a = A[i+1];
        rarr[i] = Math.max(rarr[i+1]+a, a, 0)
    }

    
    // console.log(rarr)
    let max = 0;
    for(let i = 1 ; i < n-1 ; i++){
        const a =larr[i]+rarr[i];
        if(a>max){
            max = a;
        }
    }
    return max;
}

File: codility/python/MaxDoubleSliceSum.py
def solution(A):
    # Implement your solution here
    n = len(A) 
    sum1 = [0] * n
    sum2 = [0] * n
    sum1[1] = 0
    sum2[n-2] = 0
    for i in range(2, n-1):
        if sum1[i-1] < 0:
            sum1[i] = A[i-1]
        else:
            sum1[i] = sum1[i-1]+A[i-1]

    for i in range(n-3, 0, -1):
        if sum2[i+1] < 0:
            sum2[i] = A[i+1]
        else:
            sum2[i] = sum2[i+1]+A[i+1]
    
    sum = [0] * n
    for i in range(n):
        sum[i] = (sum1[i] if sum1[i] > 0 else 0) + (sum2[i] if sum2[i] > 0 else 0)
    return max(sum)